# Tissue-enrichment Test for Transcriptome Datasets
This repository contains scripts that allow the user to test a transcriptome (RNA-seq, microarray)
for enrichment of transcripts specific to certain tissue types. By default the test is set up for
spatiotemporal specific gene expression in developing seeds of Arabidopsis thaliana, using the GEO
superseries [GSE12404](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE12404).
The scripts included here enable the user to:
1. generate-tissue-specific-genes.R: Use a reference gene expression atlas to identify genes that are spatially and/or temporally expressed in specific tissues or cell types.
2. tissue-enrichment-test.R: Perform a statistical enrichment test to determine which tissue types from the atlas are represented in a given transcriptome dataset.

To download these scripts, run this command in your desired destination folder:
```shell
git clone https://gitlab.com/nodine-lab/tissue-enrichment-test.git
```

After cloning the repository, run these scripts on the command line with:
```shell
cd tissue-enrichment-test/
Rscript generate-tissue-specific-genes.R
Rscript tissue-enrichment-test.R
```

These scripts require the statistical software [R](https://www.r-project.org/) to be installed on your
computer. Generating tissue-enrichment heatmaps additionally requires the R package [pheatmap](https://cran.r-project.org/web/packages/pheatmap/index.html).

To test a tissue-specific transcriptome from Arabidopsis seeds, all you will need is a tab-delimited
file of gene expression values (TPM, FPKM, or microarray probe intensities), where each row is a unique
gene name and each column is one transcriptome (see 'nodine_2012_embryos.tsv' for an example).

## 1. generate-tissue-specific-genes.R
This script is used to generate the file 'specific_genes.txt', which contains lists of genes specifically
expressed in one cell/tissue type at a certain time. You can change the statistical cutoffs by modifying
the variables defined at the top of the script.

To generate the list of tissue-specific gene expression, the script requires a reference gene expression atlas
of high quality that includes the tissue-of-interest and all nearby tissue types. In our experience,
transcriptomes generated by laser capture microdissection (LCM) give the highest specificity,
and we recommend that LCM datasets should be used as a reference if they are available.

Save your reference atlas in the working directory as 'reference_atlas.tsv'. The name of each sample must be
formatted as: timepoint_tissue.replicate (e.g. 1_EP.1 or 6_GSC.2). If you only have
spatial or temporal data, make sure that the sample names contain no underscores. Timepoints must be numeric
to allow the script to identify adjacent timepoints.

Two files are used by generate-tissue-specific-genes.R to interpret the reference atlas:
'timepoints.txt' and 'tissues.txt'. Replace the descriptions with the appropriate information. If any samples
in 'reference_atlas.tsv' should be ignored, add '//exclude' to the description of that timepoint or tissue.
In our example, whole seed samples were excluded because they do not represent a specific tissue within the seed.

At least two biological replicates are required for each tissue in the reference atlas.

## 2. tissue-enrichment-test.R
This script uses the tissue-specific gene expression calculated in step 1 to check for enrichment of each tissue
type in the transcriptomes provided. Two samples are required to test a table of transcriptomes: a .TSV file
containing the actually gene expression values, and a .DESCRIPTION file with a column for sample names and a
column for the developmental timepoint to which that sample should be compared. As examples, you can find both
'nodine_2012_embryos.TSV/.DESCRIPTION' and 'reference_atlas.TSV/DESCRIPTION'.

Some user options are available, and can be changed at the top of the file.